### PR TITLE
Add CreateSecurityGroup step for Azure provider

### PR DESCRIPTION
**Pull request by saheienko**
This step will create Azure network security groups for:
* masters (expose ssh and kube-apiserver ports for supergiant public address)
* nodes (expose ssh port for supergiant public address)

### CODE REVIEW

File: pkg/workflows/steps/azure/create_group_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package azure
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/supergiant/control/pkg/sgerrors"
+	"github.com/supergiant/control/pkg/workflows/steps"
```
Comment (stgleb) [inline pkg/workflows/steps/azure/create_group_test.go]: gofmt

File: pkg/workflows/steps/azure/create_group_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+	require.NotNil(t, s.baseClientFn, "base client shouldn't be nil")
+
+	var nilStringSlice []string
+	require.Equal(t, nil, s.Rollback(context.Background(), nil, nil), "rollback not implemented")
+	require.Equal(t, nilStringSlice, s.Depends(), "depends not implemented")
+	require.Equal(t, CreateGroupStepName, s.Name(), "check step name")
+	require.Equal(t, "Azure: Create ResourceGroup", s.Description(), "check description")
+}
+
```
Comment (stgleb) [inline pkg/workflows/steps/azure/create_group_test.go]: Maybe one more testcase?

File: pkg/workflows/steps/azure/create_group_test.go (hunk 1 above)
Comment (saheienko) [reply pkg/workflows/steps/azure/create_group_test.go]: Done.

File: pkg/workflows/steps/azure/create_group_test.go (hunk 2 above)
Comment (saheienko) [reply pkg/workflows/steps/azure/create_group_test.go]: Tests has been added.

**Pull request merged**